pkg/backend/httpstate/diffs_pre_go120.go

```diff
@@ -0,0 +1,18 @@
+//go:build !go1.20
+
+package httpstate
+
+import (
+	"encoding/json"
+	"unsafe"
+)
+
+// unsafeAsString converts the contents of the given json.RawMessage into a string without allocating. The contents
+// of the returned string will change if the json.RawMessage is mutated before the referenced string goes out of scope.
+// It is imperative that callers ensure that the lifetime of the return value does not exceed the lifetime of the
+// input contents.
+func unsafeAsString(raw json.RawMessage) string {
```

Comment:
Are we sure the slight extra perf is worth having "unsafe" in the code?

Especially since eventually we may transition to dds.computeEdits that operates on byte slices in the first place, so we can avoid dealing with strings all-together right? Perhaps just avoid unsafe and wait until we can land that? 

Up to y'all judgment. Thanks!

Reply:
I haven't fully evaluated our needs here, but FWIW, there's an alternative API where we invert control (pass a sort.Interface-style access to the storage instead of passing a slice) which would also take care of this unsafe-ness if the generic implementation doesn't already. This is the API [go-cmp's internal/diff uses](https://pkg.go.dev/github.com/google/go-cmp@v0.5.9/cmp/internal/diff).